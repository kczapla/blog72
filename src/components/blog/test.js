import React from 'react'
import axios from 'axios'

import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import Blog from './index'

jest.mock('../blog-item-list', () => {
  return {
    __esModule: true,
    default: function BlogItemList (params) {
      return <div>{params.posts.testMessage}</div>
    }
  }
})

jest.mock('axios')

describe('blog', () => {
  it('fetches posts from api', async () => {
    axios.get.mockResolvedValue({ data: { testMessage: 'this is test' } })
    render(<Blog/>)
    await expect(screen.findByText('this is test')).resolves.toBeTruthy()
  })
})
