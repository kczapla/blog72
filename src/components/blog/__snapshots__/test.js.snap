// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blog renders correctly 1`] = `
<article
  className="sc-bdfBwQ iylRfi"
>
  <div />
</article>
`;
